modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-order-impl/src/main/java/com/liferay/headless/commerce/delivery/order/internal/resource/v1_0/PlacedOrderResourceImpl.java

```diff
@@ -132,6 +121,10 @@ public Page<PlacedOrder> getChannelAccountPlacedOrdersPage(
 					channelExternalReferenceCode);
 		}
 
+		AccountEntry accountEntry =
+			_accountEntryService.getAccountEntryByExternalReferenceCode(
+				accountExternalReferenceCode, contextCompany.getCompanyId());
+
```

Comment:
also here put it back to the original place... if the account doesn't exists doesn't make sense that we check for the channels and bla bla bla...